toolchain/parser/parser_state.def

```diff
@@ -58,27 +67,23 @@ CARBON_PARSER_STATE(BraceExpressionParameterAsUnknown)
 //   2. BraceExpressionParameterFinishAs(Type|Value|Unknown)
 // Else:
 //   1. BraceExpressionParameterFinishAs(Type|Value|Unknown)
-CARBON_PARSER_STATE(BraceExpressionParameterAfterDesignatorAsType)
-CARBON_PARSER_STATE(BraceExpressionParameterAfterDesignatorAsValue)
-CARBON_PARSER_STATE(BraceExpressionParameterAfterDesignatorAsUnknown)
+CARBON_PARSER_STATE_VARIANTS3(BraceExpressionParameterAfterDesignator, Type,
```

Comment:
Can you clarify what's diverged? I'm not seeing it.

Reply:
Ah, I think this is just a misunderstanding of the comments. I think you're assuming they describe possible input states. Instead, the comments describe possible output states; the format is documented at the top of the file.

In this case, the AfterDesignator state doesn't directly repeat itself. As a consequence, regardless of whether the input is valid, it will never go to AfterDesignator. The only difference between a valid and invalid input is whether it will process an Expression. It always does a Finish.

Note that for some states, particularly Expression and Pattern, documenting input states may be really complex. It's also not always going to be clear what parts of an input stack should be documented.

Reply:
I see, "AfterDesignator" and "Finish" were just close enough that I thought they might be referring to the same thing.